framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
I like the additional constructor flag... this is something that should be decided by the designer of the System/Class that is inheriting from BlockRestrictable... and it's much more direct to pass it in the constructor.